### PR TITLE
[feature](nereids) Support basic aggregate rewrite and function rollup using materialized view

### PR DESCRIPTION
**Pull request by seawinde**
## Proposed changes

Add aggregate materializedviewRules for query rewrite.
it support the query rewrite as following:

    def mv = "select lineitem.L_LINENUMBER, orders.O_CUSTKEY, sum(O_TOTALPRICE) as sum_alias " +
            "from lineitem " +
            "inner join orders on lineitem.L_ORDERKEY = orders.O_ORDERKEY " +
            "group by lineitem.L_LINENUMBER, orders.O_CUSTKEY "
    def query = "select lineitem.L_LINENUMBER, sum(O_TOTALPRICE) as sum_alias " +
            "from lineitem " +
            "inner join orders on lineitem.L_ORDERKEY = orders.O_ORDERKEY " +
            "group by lineitem.L_LINENUMBER"

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by seawinde** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/AbstractMaterializedViewJoinRule.java [hunk 1]
```diff
@@ -40,28 +35,23 @@
  * This is responsible for common join rewriting
  */
 public abstract class AbstractMaterializedViewJoinRule extends AbstractMaterializedViewRule {
-    private static final HashSet<JoinType> SUPPORTED_JOIN_TYPE_SET =
-            Sets.newHashSet(JoinType.INNER_JOIN, JoinType.LEFT_OUTER_JOIN);
-
     @Override
     protected Plan rewriteQueryByView(MatchMode matchMode,
             StructInfo queryStructInfo,
```
Comment (XieJiann) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/AbstractMaterializedViewJoinRule.java:40]: Nullable annotion

**Review by keanji-x — APPROVED** [review]

**Comment by seawinde** [comment]
run buildall

**Comment by seawinde** [comment]
run buildall

**Comment by seawinde** [comment]
run buildall

**Comment by seawinde** [comment]
run buildall

**Comment by seawinde** [comment]
run buildall

**Comment by seawinde** [comment]
run buildall

**Comment by seawinde** [comment]
run buildall

**Comment by seawinde** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Pull request merged**